packages/backend/src/ee/services/ai/agents/tests/testUtils.ts

```diff
@@ -0,0 +1,424 @@
+    if (!config) return ['id', 'count'];
+
+    const dimensionNames = config.dimensions.map((d) => d.name);
+    const metricNames = config.metrics.map((m) => m.name);
+    return [...dimensionNames, ...metricNames];
+};
+
+export const createMockExplore = (exploreName: string): Explore => {
+    const config = EXPLORE_CONFIGS[exploreName as keyof typeof EXPLORE_CONFIGS];
+    const summary = mockExploresSummary.find((e) => e.name === exploreName);
+
+    let tables;
+
+    if (config) {
+        const dimensions = Object.fromEntries(
+            config.dimensions.map((dim) => [
+                dim.name,
+                createDimension(
+                    dim.name,
+                    dim.label,
+                    dim.description,
+                    dim.type,
+                    exploreName,
+                ),
+            ]),
+        );
+
+        const metrics = Object.fromEntries(
+            config.metrics.map((metric) => [
+                metric.name,
+                createMetric(
+                    metric.name,
+                    metric.label,
+                    metric.description,
+                    metric.sql,
+                    exploreName,
+                ),
+            ]),
+        );
+
+        tables = {
+            [exploreName]: {
+                name: exploreName,
+                label: capitalize(exploreName),
+                description: `${capitalize(exploreName)} information table`,
+                dimensions,
+                metrics,
+            },
+        };
+    } else {
+        tables = {
+            [exploreName]: {
+                name: exploreName,
+                label: capitalize(exploreName),
+                description: `${exploreName} table`,
+                dimensions: {
+                    id: createDimension(
+                        'id',
+                        'ID',
+                        `Unique identifier for ${exploreName}`,
+                        'string',
+                        exploreName,
+                    ),
+                },
+                metrics: {
+                    count: createMetric(
+                        'count',
+                        'Count',
+                        `Count of ${exploreName}`,
+                        'COUNT(*)',
+                        exploreName,
+                    ),
+                },
+            },
+        };
+    }
+
+    return {
+        name: exploreName,
+        label: capitalize(exploreName),
+        tags: [],
+        baseTable: summary?.baseTable || exploreName,
+        joinedTables: [],
+        targetDatabase: undefined,
+        tables,
+    } as unknown as Explore;
+};
+
+export const createArgs = ({
+    agentSettings,
+    messageHistory,
+    ...options
+}: Partial<Omit<AiAgentArgs, 'agentSettings'>> & {
+    agentSettings?: Partial<AiAgentArgs['agentSettings']>;
+} = {}) => {
+    const { OPENAI_API_KEY, OPENAI_MODEL_NAME = 'gpt-4o' } = process.env;
```

Comment:
I feel like we are dragging this env variable a lot. should this be something that you pass  to createArgs maybe?

Reply:
created a factory to solve this